File: backend/app/schemas/retention.py
from __future__ import annotations

from pydantic import BaseModel, Field


class RetentionSettingsOut(BaseModel):
    retention_days_finished: int
    retention_days_rejected: int
    retention_days_failed: int


class RetentionSettingsUpdate(BaseModel):
    retention_days_finished: int | None = Field(default=None, ge=1, le=3650)
    retention_days_rejected: int | None = Field(default=None, ge=1, le=3650)
    retention_days_failed: int | None = Field(default=None, ge=1, le=3650)


class RetentionRunResult(BaseModel):
    deleted_job_count: int
    bytes_freed: int


class StorageOverviewOut(BaseModel):
    disk_total_bytes: int
    disk_used_bytes: int
    disk_free_bytes: int
    uploads_bytes: int
    sliced_bytes: int
    active_uploaded_files: int
    jobs_pending_retention: int

File: frontend/src/lib/retention.ts
export interface RetentionSettings {
  retention_days_finished: number;
  retention_days_rejected: number;
  retention_days_failed: number;
}

export interface RetentionRunResult {
  deleted_job_count: number;
  bytes_freed: number;
}

export interface StorageOverview {
  disk_total_bytes: number;
  disk_used_bytes: number;
  disk_free_bytes: number;
  uploads_bytes: number;
  sliced_bytes: number;
  active_uploaded_files: number;
  jobs_pending_retention: number;
}

export function formatBytes(bytes: number): string {
  if (bytes === 0) return "0 B";
  const units = ["B", "KB", "MB", "GB", "TB"];
  const exponent = Math.min(units.length - 1, Math.floor(Math.log(bytes) / Math.log(1024)));
  const value = bytes / 1024 ** exponent;
  return `${value.toFixed(exponent === 0 ? 0 : 1)} ${units[exponent]}`;
}
